app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java

```diff
@@ -327,21 +327,31 @@ public String unitized_string(double value) {
         }
     }
 
-    public String unitizedDeltaString(double value) {
+    public String unitizedDeltaString(boolean showUnit) {
+
+        List<BgReading> last2 = BgReading.latest(2);
+        if(last2.size() < 2 || last2.get(0).timestamp - last2.get(1).timestamp > 20 * 60 * 1000){
+            // don't show delta if there are not enough values or the values are more than 20 mintes apart
+            return "--";
+        }
+
+        double value = BgReading.currentSlope() * 5*60*1000;
+
         DecimalFormat df = new DecimalFormat("#");
-        df.setMaximumFractionDigits(1);
         String delta_sign = "";
-        if (value > 0.1) { delta_sign = "+"; }
+        if (value > 0) { delta_sign = "+"; }
         if(doMgdl) {
-            return delta_sign + df.format(unitized(value)) + " mg/dl";
+            df.setMaximumFractionDigits(1);
+            return delta_sign + df.format(unitized(value)) +  (showUnit?" mg/dl":"");
         } else {
+            df.setMaximumFractionDigits(2);
```

Comment:
And I believe when the pebble is dealing with mmol it wants just one decimal point